libtenzir/builtins/commands/exec.cpp

```diff
@@ -91,6 +92,80 @@ auto add_implicit_source_and_sink(pipeline pipe) -> caf::expected<pipeline> {
   return pipe;
 }
 
+auto format_metric(const metric& metric) {
+  auto result = std::string{};
+  auto it = std::back_inserter(result);
+  const auto locale = std::locale("en_US.UTF-8");
+  constexpr auto indent = std::string_view{"  "};
+  it = fmt::format_to(it, "operator #{} ({})\n", metric.operator_index + 1,
+                      metric.operator_name);
+  it = fmt::format_to(it, "{}elapsed: {}\n", indent, data{metric.time_elapsed});
+  it = fmt::format_to(it, "{}scheduled: {} ({:.2f}%)\n", indent,
+                      data{metric.time_scheduled},
+                      100.0 * static_cast<double>(metric.time_scheduled.count())
+                        / static_cast<double>(metric.time_elapsed.count()));
+  it = fmt::format_to(it, "{}running: {} ({:.2f}%)\n", indent,
+                      data{metric.time_running},
+                      100.0 * static_cast<double>(metric.time_running.count())
+                        / static_cast<double>(metric.time_elapsed.count()));
+  if (metric.inbound_measurement.unit != "void") {
+    it = fmt::format_to(it, "{}inbound:\n", indent);
+    it = fmt::format_to(
+      it, locale, "{}{}{}: {:L} at a rate of {:.2f}/s\n", indent, indent,
```

Comment:
```suggestion
      it, locale, "{0}{0}{1}: {2:L} at a rate of {3:.2f}/s\n", indent,
```

Would using indexes in this function result in more readable format strings? Up to you.

Reply:
While I agree it's more readable I don't think it's worth changing right now. If we touch it again then maybe we can rewrite it.